js/common/util/compute_domain.js

```diff
@@ -0,0 +1,9 @@
+import { extractDomain } from 'home-assistant-js-websocket';
```

Comment:
This is problematic. We are individually packaging `home-assistant-js-websocket` and making it available on the window as `window.HAWS`. I however have come to realize that we should extract the helpers from the websocket package and just move them to the polymer repo.

Reply:
The roadmap I had pictured is that we'd gradually reduce dependence on our global state, and import functions from the ws lib where they're needed.

Reply:
What do you think about explicit imports here, so that we can gradually import pieces where they're used, and remove them from the global `HAWS`. It'll help reduce the size of `core.js`.

https://github.com/home-assistant/home-assistant-polymer/blob/830121ec70f3c8f0ee860e12cba84841bf5a72af/js/core.js#L1

Reply:
I think it's better to just copy the methods that we want from HAWS to this repo. We can copy the tests too.

Once all dependencies are gone, I'll release a 2.0 with those methods removed.

There are not too many:
<img width="910" alt="image" src="https://user-images.githubusercontent.com/1444314/32987542-b8acff64-cca3-11e7-8673-77d7e9ed829e.png">